src/main/java/se/sundsvall/billingdatacollector/integration/party/PartyClient.java

```diff
@@ -6,22 +6,24 @@
 
 import java.util.Optional;
 
+import org.springframework.cache.annotation.Cacheable;
 import org.springframework.cloud.openfeign.FeignClient;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 
 import generated.se.sundsvall.party.PartyType;
 
 @FeignClient(
-    name = CLIENT_ID,
-    configuration = PartyIntegrationConfiguration.class,
-    url = "${integration.party.base-url}"
+	name = CLIENT_ID,
+	configuration = PartyIntegrationConfiguration.class,
+	url = "${integration.party.base-url}"
 )
 interface PartyClient {
 
-    @GetMapping(
-        path = "/{type}/{legalId}/partyId",
-        produces = { TEXT_PLAIN_VALUE, APPLICATION_PROBLEM_JSON_VALUE }
-    )
-    Optional<String> getPartyId(@PathVariable("type") PartyType partyType, @PathVariable("legalId") String legalId);
+	@GetMapping(
+		path = "/{type}/{legalId}/partyId",
+		produces = { TEXT_PLAIN_VALUE, APPLICATION_PROBLEM_JSON_VALUE }
+	)
+	@Cacheable("partyId")
+	Optional<String> getPartyId(@PathVariable("type") PartyType partyType, @PathVariable("legalId") String legalId);
```

Comment:
I'm not sure, but I think that if `Optional` is to be returned when legalId can't be translated (is not found), a `dismiss404 = true` is needed in the `@FeignClient` annotation.